types/key/node.go

```diff
@@ -265,6 +265,14 @@ func (k NodePublic) Less(other NodePublic) bool {
 	return bytes.Compare(k.k[:], other.k[:]) < 0
 }
 
+// Equal reports whether k is bytewise equal to an other NodePublic.
+//
+// Not that this function can return false for two different encodings
+// of the same public key.
+func (k NodePublic) Equal(other NodePublic) bool {
```

Comment:
No need to define Equal methods for `FooPublic` types, they are deliberately designed to be comparable as value types, so you can use `==` and get the right result. `Equal` method exist on private keys to enforce constant-time comparison (and the `FooPrivate` struct definitions contain a hack that make the structs incomparable using `==`).